lte/protos/session_manager.proto

```diff
@@ -540,3 +556,226 @@ service CentralSessionController {
+  bool pdu_session_status_notification_sub = 13;   //Subscription For PDU Session Status Notification true or false
+  DnnSelectionMode dnn_selection_mode = 14;       // TS-29-502 - 6.1.6.3.10
+  string trace_requirment = 15;                   // TS-29.571 5.6.
+  bytes integrity_protection_max_data_rate = 16;  //Length is 3 bytes, TS-24.501 9.11.4.7
+  PduSessionType pdu_session_type = 17;
+  SscMode ssc_mode = 18;
+  M5GSMCapability m5g_sm_capability = 19;  // max size go up to 15 bytes. TLV and 9.11.4.1.
+  uint32 max_pkt_filter_supported = 20;    //Maximum no. of supported packet filters
+  bool pdu_session_req_always_on = 21;     //Always-on PDU session requested TS-24.501 9.11.4.4
+  string sm_pdu_dn_req = 22;               //SM PDU DN request container. Max size up to 255. TLV and 9.11.4.16.
+  M5GSMCause m5gsm_cause = 23;
+  QosRules requested_qos_rules = 24; //TS-24501 - 9.11.4.13 minimum 7 and max 65538, variable length.
+}
+
+message SetSMSessionContext {
+  CommonSessionContext common_context = 1;  //Common message for 4g, 5g and WiFi.
+  RatSpecificContext   rat_specific_context = 2;
+}
+message SmContextVoid {
+
+  string output = 1;
+
+}
+
+
+//SET API for 3 procedures request from AMF to SMF
+        service AmfPduSessionSmContext {
+                rpc SetAmfSessionContext (SetSMSessionContext) returns (SmContextVoid);
+        }
+
+////////// SET MESSAGE PROTO DEFINATION OF SMF TO AMF ON RESPONSE /////////////
+
+//4G response message has not been refactored yet. Hence only considering M5G.
+message RatSpecificContextAccess {
+  //oneof context {
+    //LTESessionContextAccess lte_context = 1;
+    M5GSMSessionContextAccess m5g_session_context_rsp = 2;
+  //}
+}
+
+//AuthorizedQosRules
+//The purpose of the QoS rules information element is to indicate a set of QoS rules to be used by the UE, 
+//where each QoS rule is a set of parameters.
+//a) for classification and marking of uplink user traffic; and
+//b) for identification of a QoS flow which the network is to use for a particular downlink user traffic.
+//As the UE needs to be aware of a QoS flow which the network is to use for a particular downlink user traffic
+//e.g. to determine whether a resource is available for downlink media of a media stream of an SDP media
+//description provided by the UE in an IMS session.
+
+message M5GSMSessionContextAccess{
+    //One byte, received from UE. For 5G pdu_session_id. 
+    bytes pdu_session_id = 1;
+    //selected PDU session TS-24501 - 9.11.4.11
+    PduSessionType    pdu_session_type = 2;
+    //selected session and service continuty mode TS-24501 - 9.11.4.16
+    SscMode selected_ssc_mode = 3; 
+    //TS-24501 - 9.11.4.3 - minimum 7 and max 65538, variable length
+    repeated QosRules authorized_qos_rules = 4; 
+    //TS-24501 - 9.11.4.14 PDU session Aggregate Maximum Bit Rate imported from subscriberdb.proto
+    AggregatedMaximumBitrate session_ambr = 5;
```

Comment:
Where does session_ambr come from in this case?

Reply:
AMBR provided by AMF. Hence only declaration in proto. Not implemented in SessionD.

Reply:
If it comes from AMF, then why is it in this message? I thought this message was SMF->AMF

Reply:
In-fact, AMBR calculation depends on AMF's input and respective policy of UE. Hence in some situation, AMBR may defer from what AMF have provided and the same will be passed to UE by AMF. 
Hope I might have clear this doubt.

Reply:
So the AMBR value depends on policies that are returned by PCF? I'm not sure if I'm understanding this correctly. What exactly would be the logic required in SMF to construct this value?

Reply:
Yes, you are correct "AMBR value depends on policies that are returned by PCF" . Even I had discussed with AMF team Sandeep and confirmed.

Reply:
Ok, I understand.